src/Fantomas.Tests/ControlStructureTests.fs

```diff
@@ -836,3 +836,35 @@ with ex ->
     Infrastructure.ReportWarning ex
     return None
 """
+
+[<Test>]
+let ``MultilineBlockBracketsOnSameColumn should be honored inside match block, 1238`` () =
+    formatSourceString
+        false
+        """
+      module Foo =
+          let Bar () =
+              if x then
+                  match foo with
+                  | { Bar = true
+                      Baz = _ } -> failwith "xxx"
+                  | _ -> None
+"""
+        { config with
+              MaxLineLength = 30
+              MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+module Foo =
+    let Bar () =
+        if x then
+            match foo with
+            | {
+                  Bar = true
+                  Baz = _
+              }
+                -> failwith "xxx"
```

Comment:
Multiline match patterns do not put the arrow on the next line as far as I know.
For example:
```fsharp
match t.Type with
| TriviaNodeType.Token (_, tok) when
    (range.StartLine = t.Range.StartLine
     && range.StartColumn < t.Range.StartColumn) -> Some(tok, t)
| _ -> None
```
so I would not make an exception for this unless there is clear guidance from a style guide.

Reply:
I have changed the expect result of the test according multiline match patterns.